apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import TraceKit from 'tracekit';
+
+/**
+ * Internal dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Module variables
+ */
+
+/**
+ * Interval for error reports so we don't flood te endpoint. More frequent
+ * reports get throttled.
+ *
+ * @type {number}
+ */
+const REPORT_INTERVAL = 60000;
+
+const diagnosticData = {
+	user_id: 'todo-get-the-user-id',
+	blog_id: 'todo-get-the-blog-id',
+	extra: {
+		throttled: 0,
+	},
+};
+
+let lastReport = 0;
+
+if ( ! window.onerror ) {
+	TraceKit.report.subscribe( ( errorReport ) => {
+		debugger;
+		const error = {
+			message: errorReport.message,
+			url: document.location.href,
+		};
+
+		if ( Array.isArray( errorReport.stack ) ) {
+			const trace = errorReport.stack.slice( 0, 10 );
+			trace.forEach( ( report ) =>
```

Comment:
I haven't dug into this portion yet and I'm not entirely sure what it does / if it's needed. At first glance, it seems to truncate the fields.

Reply:
I wouldn't even use tracekit for this first version. Just the most basic get the error and store it.

Reply:
Sounds good 👍🏻